### PR TITLE
feat!: better query parsing

### PR DESCRIPTION
**Pull request by barakalon**
This brings on sqlglot as a dependency, which lets us add more built-in handling of MySQL specific things.

Mysql-mimic now handles INFORMATION_SCHEMA, as well as a slew of other strange queries I encountered while testing, like `show /*!50002 full*/ tables`

### CODE REVIEW

File: mysql_mimic/intercept.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        return exp.FALSE.copy()
+    if value is None:
+        return exp.NULL.copy()
+    if isinstance(value, (int, float)):
+        return exp.Literal.number(value)
+    return exp.Literal.string(str(value))
+
+
+def expression_to_value(expression: exp.Expression) -> Any:
+    if expression == exp.TRUE:
```
Comment (tobymao) [inline mysql_mimic/intercept.py:33]: i guess this is ok to use the singletons for just checks?

**Review by tobymao — APPROVED** [review]

File: mysql_mimic/intercept.py (hunk 1 above)
Comment (barakalon) [reply mysql_mimic/intercept.py]: Ah, nice catch. Been working on this one for a while, since before we added those new functions.

Comment (barakalon) [reply mysql_mimic/intercept.py]: (I think it's better to use the functions, even if its slightly less efficient?)

Comment (tobymao) [reply mysql_mimic/intercept.py]: yea

**Review by kelsin — APPROVED** [review]

**Pull request merged**